addons/hr_timesheet/views/hr_timesheet_portal_templates.xml

```diff
@@ -50,6 +50,7 @@
                     <t t-foreach="grouped_timesheets" t-as="timesheets_with_hours">
                         <t t-set="timesheets" t-value="timesheets_with_hours[0]"/>
                         <t t-set="hours_spent" t-value="timesheets_with_hours[1]"/>
+                        <t t-set="total_hour" t-value="timesheets_with_hours[2] if len(timesheets_with_hours) > 2 else 0"/>
```

Comment:
Instead of checking the length , we should pass '0' from the controller,
```suggestion
                        <t t-set="allocated_hours" t-value="timesheets_with_hours[2]"/>
```